core/arch/arm/plat-imx/registers/imx7-gpc_regs.h

```diff
@@ -0,0 +1,353 @@
+#define BM_GPC_PU_PGC_SW_PDN_REQ_PCIE_PHY_SW_PDN_REQ		\
+			BIT32(BP_GPC_PU_PGC_SW_PDN_REQ_PCIE_PHY_SW_PDN_REQ)
+#define BP_GPC_PU_PGC_SW_PDN_REQ_USB_OTG1_PHY_SW_PDN_REQ	(2)
+#define BM_GPC_PU_PGC_SW_PDN_REQ_USB_OTG1_PHY_SW_PDN_REQ	\
+			BIT32(BP_GPC_PU_PGC_SW_PDN_REQ_USB_OTG1_PHY_SW_PDN_REQ)
+#define BP_GPC_PU_PGC_SW_PDN_REQ_USB_OTG2_PHY_SW_PDN_REQ	(3)
+#define BM_GPC_PU_PGC_SW_PDN_REQ_USB_OTG2_PHY_SW_PDN_REQ	\
+			BIT32(BP_GPC_PU_PGC_SW_PDN_REQ_USB_OTG2_PHY_SW_PDN_REQ)
+#define BP_GPC_PU_PGC_SW_PDN_REQ_USB_HSIC_PHY_SW_PDN_REQ	(4)
+#define BM_GPC_PU_PGC_SW_PDN_REQ_USB_HSIC_PHY_SW_PDN_REQ	\
+			BIT32(BP_GPC_PU_PGC_SW_PDN_REQ_USB_HSIC_PHY_SW_PDN_REQ)
+
+#define BP_GPC_CPU_PGC_SW_PUP_STATUS_CORE0_A7_SW_PUP_STATUS	(0)
+#define BM_GPC_CPU_PGC_SW_PUP_STATUS_CORE0_A7_SW_PUP_STATUS	\
+		BIT32(BP_GPC_CPU_PGC_SW_PUP_STATUS_CORE0_A7_SW_PUP_STATUS)
+#define BP_GPC_CPU_PGC_SW_PUP_STATUS_CORE1_A7_SW_PUP_STATUS	(1)
+#define BM_GPC_CPU_PGC_SW_PUP_STATUS_CORE1_A7_SW_PUP_STATUS	\
+		BIT32(BP_GPC_CPU_PGC_SW_PUP_STATUS_CORE1_A7_SW_PUP_STATUS)
+#define BP_GPC_CPU_PGC_SW_PUP_STATUS_SCU_A7_SW_PUP_STATUS	(2)
+#define BM_GPC_CPU_PGC_SW_PUP_STATUS_SCU_A7_SW_PUP_STATUS	\
+		BIT32(BP_GPC_CPU_PGC_SW_PUP_STATUS_SCU_A7_SW_PUP_STATUS)
+
+#define BP_GPC_CPU_PGC_SW_PDN_STATUS_CORE0_A7_SW_PDN_STATUS	(0)
+#define BM_GPC_CPU_PGC_SW_PDN_STATUS_CORE0_A7_SW_PDN_STATUS	\
+		BIT32(BP_GPC_CPU_PGC_SW_PDN_STATUS_CORE0_A7_SW_PDN_STATUS)
+#define BP_GPC_CPU_PGC_SW_PDN_STATUS_CORE1_A7_SW_PDN_STATUS	(1)
+#define BM_GPC_CPU_PGC_SW_PDN_STATUS_CORE1_A7_SW_PDN_STATUS	\
+		BIT32(BP_GPC_CPU_PGC_SW_PDN_STATUS_CORE1_A7_SW_PDN_STATUS)
+#define BP_GPC_CPU_PGC_SW_PDN_STATUS_SCU_A7_SW_PDN_STATUS	(2)
+#define BM_GPC_CPU_PGC_SW_PDN_STATUS_SCU_A7_SW_PDN_STATUS	\
+		BIT32(BP_GPC_CPU_PGC_SW_PDN_STATUS_SCU_A7_SW_PDN_STATUS)
+
+
+/*
+ * GPC Registers - PGC
+ */
+#define GPC_PGC_A7CORE0_CTRL	0x800
+#define GPC_PGC_A7CORE0_PUPSCR	0x804
+#define GPC_PGC_A7CORE0_PDNSCR	0x808
+#define GPC_PGC_A7CORE1_CTRL	0x840
+#define GPC_PGC_A7CORE1_PUPSCR	0x844
+#define GPC_PGC_A7CORE1_PDNSCR	0x848
+#define GPC_PGC_A7SCU_CTRL	0x880
+#define GPC_PGC_SCU_AUX_SW	0x890
+#define GPC_PGC_FASTMEGA_MIX	0xA00
+
+
+#define BP_GPC_PGC_nCTRL_PCR	(0)
+#define BM_GPC_PGC_nCTRL_PCR	BIT32(BP_GPC_PGC_nCTRL_PCR)
+
+#define BP_GPC_PGC_nPUPSCR_SW	(1)
+#define BM_GPC_PGC_nPUPSCR_SW	(0x3F << BP_GPC_PGC_nPUPSCR_SW)
+#define BP_GPC_PGC_nPUPSCR_PUP_WAIT_SCALL_OUT		(6)
+#define BM_GPC_PGC_nPUPSCR_PUP_WAIT_SCALL_OUT		\
+		BIT32(BP_GPC_PGC_nPUPSCR_PUP_WAIT_SCALL_OUT)
+#define BP_GPC_PGC_nPUPSCR_SW2ISO			(7)
+#define BM_GPC_PGC_nPUPSCR_SW2ISO			\
+		(0xFFFF << BP_GPC_PGC_nPUPSCR_SW2ISO)
+#define BP_GPC_PGC_nPUPSRC_SCALL_SCALLOUT_CNT		(23)
+#define BM_GPC_PGC_nPUPSCR_PUP_SCALL_SCALLOUT_CNT	\
+		(0x1FF << BP_GPC_PGC_nPUPSCR_PUP_SCALL_SCALLOUT_CNT)
+
+#define BP_GPC_PGC_SCU_AUX_SW_DFTRAM_TRC1_TMC_TMR_TCD2	(0)
+#define BP_GPC_PGC_SCU_AUX_SW_L2RETN_TRC1_TMC_TMR	(10)
+#define BP_GPC_PGC_SCU_AUX_SW_MEMPWR_TRC1_TMC_TMR	(20)
+
```

Comment:
No add unused macros.

Reply:
i would rather not, but if you insist we will remove the defined that are not being used.
however we will keep the name. is that ok with you?